internal/pkg/cli/svc_show.go

```diff
@@ -38,47 +37,53 @@ type showSvcOpts struct {
 	w             io.Writer
 	store         store
 	describer     describer
-	ws            wsSvcReader
 	sel           configSelector
-	initDescriber func(bool) error // Overriden in tests.
+	initDescriber func() error // Overriden in tests.
 }
 
 func newShowSvcOpts(vars showSvcVars) (*showSvcOpts, error) {
 	ssmStore, err := config.NewStore()
 	if err != nil {
-		return nil, fmt.Errorf("connect to environment datastore: %w", err)
+		return nil, fmt.Errorf("connect to config store: %w", err)
 	}
-	ws, err := workspace.New()
+	deployStore, err := deploy.NewStore(ssmStore)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("connect to deploy store: %w", err)
 	}
 
 	opts := &showSvcOpts{
 		showSvcVars: vars,
 		store:       ssmStore,
-		ws:          ws,
 		w:           log.OutputWriter,
 		sel:         selector.NewConfigSelect(vars.prompt, ssmStore),
 	}
-	opts.initDescriber = func(enableResources bool) error {
+	opts.initDescriber = func() error {
 		var d describer
 		svc, err := opts.store.GetService(opts.AppName(), opts.svcName)
 		if err != nil {
 			return err
 		}
 		switch svc.Type {
 		case manifest.LoadBalancedWebServiceType:
-			if enableResources {
-				d, err = describe.NewWebServiceDescriberWithResources(opts.AppName(), opts.svcName)
-			} else {
-				d, err = describe.NewWebServiceDescriber(opts.AppName(), opts.svcName)
-			}
+			d, err = describe.NewWebServiceDescriber(describe.NewWebServiceConfig{
+				NewServiceConfig: describe.NewServiceConfig{
+					App:         opts.AppName(),
+					Svc:         opts.svcName,
+					ConfigStore: ssmStore,
+				},
```

Comment:
Maybe we can initialize this into its own variable before the switch - to make it shorter.

Reply:
yeah agree on that. I'll do that in next pr. i really wish golang could support generic soon so that maybe we can do
```
option := {
    NewServiceConfig: {
		App:         opts.AppName(),
		Svc:         opts.svcName,
		ConfigStore: ssmStore,
	},
	DeployStore:     deployStore,
	EnableResources: opts.shouldOutputResources,
}
...
d, err = describe.NewWebServiceDescriber(option(describe.NewWebServiceConfig))
```